plugins/registry.js

```diff
@@ -105,7 +105,9 @@ export default class PluginRegistry {
     // - action - a function called when the button is clicked, passed the channel and channel member as arguments
     // - dropdown_text - string or React element shown for the dropdown button description
     // - tooltip_text - string shown for tooltip appear on hover
-    registerChannelHeaderButtonAction(icon, action, dropdownText, tooltipText) {
+    // - show - a function that returns true if button should be shown and false if not
```

Comment:
Independently of Kanban, this is a terrific change. Should we pass some minimum context to the new functions, such as the `channel` object or some such, to avoid having to look up this data all the time?